File: c/Image.c
#include "Image.h"

Image *image_init(uint32_t width, uint32_t height)
{
    Image *result = (Image*)malloc(sizeof(Image));
    result->width = width;
    result->height = height;
    result->capacity = width * height;
    result->buffer = (Color*)malloc(sizeof(Color) * result->capacity);
    return result;
}

void image_destroy(Image *img)
{
    free(img->buffer);
    free(img);
}

Color image_colorAt(const Image *img, int row, int col)
{
    assert(0 <= row && row < (int)img->height);
    assert(0 <= col && col < (int)img->width);
    return img->buffer[(((img->height - 1) - row) * img->width) + col];
}

void image_fill(Image *img, Color c)
{
    for(uint32_t index = 0; index < img->capacity; ++index)
        img->buffer[index] = c;
}

void image_setPixel(Image *img, int row, int col, Color c)
{
    assert(0 <= row && row < (int)img->height);
    assert(0 <= col && col < (int)img->width);
    img->buffer[(((img->height-1) - row) * img->width) + col] = c;
}

File: py/Image.py
from Color import Color

import struct

class Image:

    def __init__(self, width, height):
        self._width = width
        self._height = height
        self._capacity = width * height
        self._buffer = [Color(0, 0, 0, 255)] * self._capacity

    def width(self):
        return self._width

    def height(self):
        return self._height

    def at(self, row, col):
        assert(0 <= row and row < int(self._height));
        assert(0 <= col and col < int(self._width));
        return self._buffer[(((self._height - 1) - row) * self._width) + col];

    def set(self, row, col, color):
        assert(0 <= row and row < int(self._height));
        assert(0 <= col and col < int(self._width));
        self._buffer[(((self._height-1) - row) * self._width) + col] = color;

    def fill(self, color):
        for c in self._buffer:
            c._color = color._color

    def write(self, outFile):
        tmpBuffer = bytes("", encoding="latin_1")
        for c in self._buffer:
            outFile.write(struct.pack("<I", c._color))
